flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -517,9 +531,56 @@ void EmitLLVMAction::ExecuteAction() {
   return;
 }
 
+void EmitLLVMBitcodeAction::ExecuteAction() {
+  CompilerInstance &ci = this->instance();
+  // Generate an LLVM module if it's not already present (it will already be
+  // present if the input file is an LLVM IR/BC file).
+  if (!llvmModule_)
+    GenerateLLVMIR();
+
+  ModulePassManager MPM;
+  ModuleAnalysisManager MAM;
+
+  // Create `Target`
+  std::string error;
+  std::string theTriple = llvmModule_->getTargetTriple();
+  const llvm::Target *theTarget =
+      TargetRegistry::lookupTarget(theTriple, error);
+  assert(theTarget && "Failed to create Target");
+
+  // Create `TargetMachine`
+  std::unique_ptr<TargetMachine> TM;
+
+  TM.reset(theTarget->createTargetMachine(
+      theTriple, /*CPU=*/"", /*Features=*/"", llvm::TargetOptions(), None));
+  llvmModule_->setDataLayout(TM->createDataLayout());
+  assert(TM && "Failed to create TargetMachine");
+
+  PassBuilder PB(TM.get());
+  PB.registerModuleAnalyses(MAM);
+
+  // Generate an output file
+  std::unique_ptr<llvm::raw_ostream> os = ci.CreateDefaultOutputFile(
+      /*Binary=*/true, /*InFile=*/GetCurrentFileOrBufferName(), "bc");
+  if (!os) {
+    unsigned diagID = ci.diagnostics().getCustomDiagID(
+        clang::DiagnosticsEngine::Error, "failed to create the output file");
+    ci.diagnostics().Report(diagID);
+    return;
+  }
+
+  MPM.addPass(BitcodeWriterPass(*os));
+  MPM.run(*llvmModule_, MAM);
+
+  return;
```

Comment:
Is this `return;` needed because of some coding guideline?

Reply:
No. I'll delete it.